src/middlewares/url-validator.ts

```diff
@@ -0,0 +1,19 @@
+import { Request, Response, NextFunction } from 'express';
+import { postUrlSchema, updateUrlSchema } from '../validators/url-validator';
+
+export const validatePostUrl = async (req: Request, res: Response, next: NextFunction) => {
+  const { error, value } = postUrlSchema.validate(req.body);
+  if (error) {
+    return res.status(400).send({ message: error.details[0].message });
+  }
+  req.body = value;
+  return next();
```

Comment:
When we have more validations, we can't keep repeating this block. Extract and reuse that

Reply:
Done